src/controllers/connections.controller.ts

```diff
@@ -59,5 +60,45 @@ export class ConnectionsController {
 			commands.executeCommand(vsCommands.showErrorMessage, notification);
 			LoggerService.error(namespaces.connectionsController, log);
 		}
+
+		eventEmitter.on(`connection.${connectionInit.connectionId}.updated`, () =>
```

Comment:
Either use once or remove the subscription once you're done, otherwise you'd start getting events multiple times

And anyhow if user closes the window, unsubscribe so stuff won't happen for nothing.

Reply:
done